android/KMEA/app/src/main/java/com/keyman/engine/KeyboardPickerActivity.java

```diff
@@ -252,8 +253,17 @@ protected void onResume() {
       adapter.notifyDataSetChanged();
     }
 
-    int curKbPos = KeyboardController.getInstance().getKeyboardIndex(KMKeyboard.currentKeyboard());
-    setSelection(curKbPos);
+
+    // Determine the index to the current keyboard position to highlight as the selected keyboard
+    int currentKeyboardIndex; 
+    String currentKeyboardKey = KMKeyboard.currentKeyboard();
+    if (currentKeyboardKey == null) {
+      SharedPreferences prefs = this.getSharedPreferences(this.getString(R.string.kma_prefs_name), Context.MODE_PRIVATE);
+      currentKeyboardIndex = prefs.getInt(KMManager.KMKey_UserKeyboardIndex, 0);
```

Comment:
What if the pref has an invalid index? Could this ever happen?

Reply:
The preference should only save valid indexes. 
If the preference key doesn't exist, it'll fallback to 0.